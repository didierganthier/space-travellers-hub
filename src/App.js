import React from 'react';
import { Routes, Route } from 'react-router-dom';
import Navbar from './components/Navbar';
import NotFound from './components/NotFound';
import Missions from './pages/Missions';
import Profile from './pages/Profile';
import Rockets from './pages/Rockets';

function App() {
  return (
    <>
      <Navbar />
      <Routes>
        <Route path="/" element={<Rockets />} />
        <Route path="/Missions" element={<Missions />} />
        <Route path="/MyProfile" element={<Profile />} />
        <Route path="*" element={<NotFound />} />
      </Routes>
    </>
  );
}

export default App;
