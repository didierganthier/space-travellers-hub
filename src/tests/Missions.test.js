import React from 'react';
import {
  fireEvent, render, getByText,
} from '@testing-library/react';
import '@testing-library/jest-dom';

import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import App from '../App';
import providerWrapper from '../services/testWrapper';
import store from '../redux/configureStore';

describe('Missions page ', () => {
  test('renders missions', async () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>,
    );
    expect(getByText(tree.container, 'Missions')).toBeInTheDocument();
  });

  test('navigate to missions', async () => {
    const { getByText } = providerWrapper(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
      { store },
    );
    fireEvent.click(getByText('Missions'));
    expect(getByText('Description')).toBeInTheDocument();
  });

  test('join mission', async () => {
    const { getByText } = providerWrapper(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
      { store },
    );
    fireEvent.click(getByText('Missions'));
    setTimeout(() => {
      fireEvent.click(getByText('Join Mission'));
      expect(getByText('Leave Mission')).toBeInTheDocument();
    }, 1000);
  });

  test('Active member', async () => {
    const { getByText } = providerWrapper(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
      { store },
    );
    fireEvent.click(getByText('Missions'));
    setTimeout(() => {
      fireEvent.click(getByText('Join Mission'));
      expect(getByText('Active Member')).toBeInTheDocument();
    }, 1000);
  });
});
