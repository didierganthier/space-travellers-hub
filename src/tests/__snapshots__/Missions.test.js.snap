// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions page  renders missions 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="h-16 shadow flex justify-between px-[20px]"
      >
        <div
          class="icon-continer flex items-center h-16 my-[4px]"
        >
          <img
            alt="logo-img"
            class="w-10 h-10"
            src="https://cdn-icons-png.flaticon.com/512/3212/3212567.png"
          />
          <h1
            class="nav-text"
          >
            Space Traveler's Hub
          </h1>
        </div>
        <div
          class="options-continer flex items-center list-none gap-4"
        >
          <li>
            <a
              aria-current="page"
              class="underline"
              href="/"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              href="/Missions"
            >
              Missions
            </a>
            <span
              class="pl-4"
            >
              |
            </span>
          </li>
          <li>
            <a
              href="/MyProfile"
            >
              My Profile
            </a>
          </li>
        </div>
      </nav>
      <div />
    </div>
  </body>,
  "container": <div>
    <nav
      class="h-16 shadow flex justify-between px-[20px]"
    >
      <div
        class="icon-continer flex items-center h-16 my-[4px]"
      >
        <img
          alt="logo-img"
          class="w-10 h-10"
          src="https://cdn-icons-png.flaticon.com/512/3212/3212567.png"
        />
        <h1
          class="nav-text"
        >
          Space Traveler's Hub
        </h1>
      </div>
      <div
        class="options-continer flex items-center list-none gap-4"
      >
        <li>
          <a
            aria-current="page"
            class="underline"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            href="/Missions"
          >
            Missions
          </a>
          <span
            class="pl-4"
          >
            |
          </span>
        </li>
        <li>
          <a
            href="/MyProfile"
          >
            My Profile
          </a>
        </li>
      </div>
    </nav>
    <div />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
